tests/src/test/java/io/smallrye/context/test/CompletableFutureTest.java

```diff
@@ -0,0 +1,177 @@
+                    Assert.assertNull(MyContext.get());
+                    return v;
+                })
+                .get();
+        // nothing in the executor thread
+        managedExecutor.completedFuture(null)
+                .thenApplyAsync(v -> {
+                    Assert.assertNull(MyContext.get());
+                    return v;
+                })
+                .get();
+        // still no context
+        Assert.assertNull(MyContext.get());
+
+        // now with context
+        MyContext ctx = new MyContext();
+        MyContext.set(ctx);
+        Assert.assertEquals(ctx, MyContext.get());
+        // context in the current thread
+        managedExecutor.completedFuture(null)
+                .thenApply(v -> {
+                    Assert.assertEquals(ctx, MyContext.get());
+                    return v;
+                })
+                .get();
+        // context in the executor thread
+        managedExecutor.completedFuture(null)
+                .thenApplyAsync(v -> {
+                    Assert.assertEquals(ctx, MyContext.get());
+                    return v;
+                })
+                .get();
+        // still with context
+        Assert.assertEquals(ctx, MyContext.get());
+
+        // now create a CF while we have a context
+        CompletableFuture<Void> cfCreatedWithContext = managedExecutor.completedFuture(null);
+        // remove the context
+        MyContext.clear();
+
+        // check that we get no context, since we're capturing at lambda creation time
+        // nothing in the current thread
+        cfCreatedWithContext
+                .thenApply(v -> {
+                    Assert.assertNull(MyContext.get());
+                    return v;
+                })
+                .get();
+        // nothing in the executor thread
+        cfCreatedWithContext
+                .thenApplyAsync(v -> {
+                    Assert.assertNull(MyContext.get());
+                    return v;
+                })
+                .get();
+
+        // now set it to another context and check we get the second context
+        MyContext ctx2 = new MyContext();
+        MyContext.set(ctx2);
+        // context in the current thread
+        cfCreatedWithContext
+                .thenApply(v -> {
+                    Assert.assertEquals(ctx2, MyContext.get());
+                    return v;
+                })
+                .get();
+        // context in the executor thread
+        cfCreatedWithContext
+                .thenApplyAsync(v -> {
+                    Assert.assertEquals(ctx2, MyContext.get());
+                    return v;
+                })
+                .get();
+    }
+
+    @Test
+    public void testJava9Methods() throws InterruptedException, ExecutionException {
+        try {
+            CompletableFuture.class.getDeclaredMethod("copy");
+        } catch (NoSuchMethodException | SecurityException e) {
+            Assume.assumeTrue(false);
```

Comment:
I know it's nitpicking but why not just `return`?
Also, a comment explaining why this is ok might be good.

Reply:
If you return the test is not marked as skipped.

Reply:
I added a comment.

Reply:
> If you return the test is not marked as skipped.

Ah, fair enough, didn't realize that.